http4k-client/websocket/src/test/kotlin/org/http4k/client/WebsocketClientTest.kt

```diff
@@ -135,4 +145,24 @@ class WebsocketClientTest {
         assertThat(received.take(4).toList(), equalTo(listOf(WsMessage("bob"), WsMessage("hello"))))
         assertThat(connected, equalTo(true))
     }
+
+    @Test
+    fun `non-blocking - send and receive in binary mode`() {
+        val queue = LinkedBlockingQueue<() -> WsMessage?>()
+        val received = generateSequence { queue.take()() }
+
+        WebsocketClient.nonBlocking(Uri.of("ws://localhost:$port/bin")) { ws ->
+            ws.onMessage { message ->
+                queue.add { message }
+            }
+            ws.onClose {
+                queue.add { null }
+            }
+            ws.send(WsMessage("hello".byteInputStream()))
+        }
+
+        val messages = received.take(4).toList()
+        assertThat(messages.all { it.body is StreamBody }, equalTo(true))
```

Comment:
Please feel free to suggest a nicer way to do this instanceOf assertion with hamkrest.

Reply:
not sure... :)